### PR TITLE
fix: breadcrumb display for ProjectTree

### LINKED ISSUE #4565: Project tree incorrectly controls the URL and breadcrumb
## Describe the bug

1. After clicking on a non-root dialog 'aaa', the url got stuck in root dialog name.
bug-tree-inconsistent-url

2. Switching to non-root dialogs got wrong breadcrumb values
image

## Browser

- [ ] Electron distribution
- [ ] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

## OS

- [ ] macOS
- [ ] Windows
- [ ] Ubuntu

### LINKED ISSUE #4575: Breadcrumb does not display correctly
## Describe the bug
breadcrumb

## Browser

- [ ] Electron distribution
- [ ] Chrome
- [ ] Safari
- [ ] Firefox
- [x] Edge

## OS

- [ ] macOS
- [x] Windows
- [ ] Ubuntu

### LINKED ISSUE #4606: Can't edit per/dialog settings
## Describe the bug
If you try to click on a root dialog, within the design explorer, it always jumps you up to the root dialog. This means you have no way of changing the settings for a given dialog without editing the file directly.

cant-select-dialog-root
 

## Version

https://github.com/microsoft/BotFramework-Composer-Nightlies/releases/tag/v1.1.1-nightly.178300.83c0804

### LINKED ISSUE #4632: Project Tree generates duplicated dialog name path
## Describe the bug
Duplicated dialog names in breadcrumb
image
And inconsistent breadcrumb path
image
image

## Browser

- [ ] Electron distribution
- [ ] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

## OS

- [ ] macOS
- [ ] Windows
- [ ] Ubuntu

### LINKED ISSUE #4658: [Minor] Clicking through child dialog in a Begin a new dialog node navigates to empty
## Describe the bug

If you create a child dialog and wire it up with the main dialog, clicking through the child dialog in the **Begin a new dialog** node navigates to nothing. (In 1.1.1 version, it navigates to the **BeginDialog** trigger of the child dialog)

## Version

[Electron 1.1.1-nightly.181088.77ad8e1](https://github.com/microsoft/BotFramework-Composer-Nightlies/releases/tag/v1.1.1-nightly.181088.77ad8e1)

## Browser

- [x] Electron distribution

## OS

- [x] Windows

## To Reproduce

Steps to reproduce the behavior as documented in this [article](https://docs.microsoft.com/en-us/composer/tutorial/tutorial-add-dialog):

1. Create a child dialog and wire it up with the main dialog.
2. Clicking the child dialog in the **Begin a new dialog** node navigates to nothing. 

## Expected behavior

In 1.1.1 version, clicking through the child dialog navigates to the **BeginDialog** trigger of the child dialog itself.

## Screenshots
image

### PR DESCRIPTION
**Pull request by beyackle**
## Description

This removes most of the breadcrumb tracking logic, replacing it with a simple "you are here" display that shows the current dialog name and trigger name. The dialog name is clickable and will take a user to the dialog's properties pane.

## Task Item

closes #4575
closes #4565
closes #4632
closes #4606 
closes #4658 

## Screenshots

image

### CODE REVIEW

**Comment by srinaath** [comment]
Clicking on an action inside a trigger should add the action to the breadcrumb. This functionality needs to be at parity with the previous composer release

File: Composer/cypress/integration/Breadcrumb.spec.ts [hunk 1]
```diff
@@ -35,15 +35,15 @@ context('breadcrumb', () => {
...
     cy.findByTestId('ProjectTree').within(() => {
       cy.findByTestId('addtodo_Dialog started').click();
     });
 
-    hasBreadcrumbItems(cy, ['__TestTodoSample', 'Dialog started']);
+    hasBreadcrumbItems(cy, ['addtodo', 'Dialog started']);
   });
 
-  it('can show action name in breadcrumb', () => {
+  it.skip('can show action name in breadcrumb', () => {
```
Comment (srinaath) [inline Composer/cypress/integration/Breadcrumb.spec.ts:46]: Can remove the skip

**Review by srinaath — CHANGES_REQUESTED** [review]

File: Composer/cypress/integration/Breadcrumb.spec.ts (hunk 1 above)
Comment (beyackle) [reply Composer/cypress/integration/Breadcrumb.spec.ts]: This has been removed; it's running all 3 tests now.

File: Composer/packages/client/src/recoilModel/undo/history.ts [hunk 2]
```diff
@@ -73,20 +69,6 @@ const checkAtomsChanged = (current: AtomAssetsMap, previous: AtomAssetsMap, atom
   return atoms.some((atom) => checkAtomChanged(current, previous, atom));
 };
 
-function navigate(next: AtomAssetsMap, projectId: string) {
```
Comment (srinaath) [inline Composer/packages/client/src/recoilModel/undo/history.ts:76]: Why is navigation deleted in Undo/Redo. This would break the functionality?

**Review by srinaath — CHANGES_REQUESTED** [review]

File: Composer/packages/client/src/recoilModel/undo/history.ts [hunk 3]
```diff
@@ -184,11 +163,11 @@ export const UndoRoot = React.memo((props: UndoRootProps) => {
   });
 
   const canUndo = () => {
-    return history.canUndo();
+    return history?.canUndo?.();
```
Comment (srinaath) [inline Composer/packages/client/src/recoilModel/undo/history.ts:166]: It should not hit the component with an empty history object. Basically only when a project is loaded it should hit here

File: Composer/packages/client/src/recoilModel/undo/history.ts [hunk 4]
```diff
@@ -162,7 +142,6 @@ export const UndoRoot = React.memo((props: UndoRootProps) => {
   ) => {
     target = mapTrackedAtomsOntoSnapshot(target, current, next, projectId);
     gotoSnapshot(target);
-    navigate(next, projectId);
```
Comment (srinaath) [inline Composer/packages/client/src/recoilModel/undo/history.ts:165]: This would break the functionality if we dont navigate on an Undo or Redo

Comment (srinaath) [reply Composer/packages/client/src/recoilModel/undo/history.ts:165]: Is there a way u can store the breadcrumbs in a botstate to avoid regression?

**Comment by srinaath** [comment]
@beyackle The breadcrumb works fine now but I see a bug where when a dialog calls another dialog its not navigating automatically to the BeginDialog trigger of this new dialog. Hence the breadcrumb is wrong in this scenario.

**Comment by beyackle** [comment]
> @beyackle The breadcrumb works fine now but I see a bug where when a dialog calls another dialog its not navigating automatically to the BeginDialog trigger of this new dialog. Hence the breadcrumb is wrong in this scenario.
> 
> 

That's already another existing issue (and one that's being tracked in our board), so I'll fix that here too and update the comment accordingly.

**Comment by srinaath** [comment]
@lei9444 check this branch out. As a part of this work the breadcrumbs state is removed. Its read from the URL. I think we wouldnt need to use that state anymore.

**Comment by lei9444** [comment]
Hi, @beyackle @srinaath 
image

for previous design, we want to show the parent dialog, if we click the test, the breadcrumbs is like
main dialog > main trigger > main action > test dialog
 
do we need to support it today?

**Comment by srinaath** [comment]
@boydc2014  this PR looks good on my end. I think once u test it u can approve and merge it

**Comment by srinaath** [comment]
> Hi, @beyackle @srinaath  
>  
> 
> image
> 
> for previous design, we want to show the parent dialog, if we click the test, the breadcrumbs is like  
> main dialog > main trigger > main action > test dialog
> 
> do we need to support it today?

Looks like we dont need to support it today

**Review by srinaath — APPROVED** [review]

**Comment by lei9444** [comment]
I have tested it, and looks good to me

**Review by srinaath — APPROVED** [review]

**Pull request merged**